Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1

```diff
@@ -90,21 +90,42 @@ function CreateCommandLineArgs
     return $sb.ToString();
 }
 
-function UpdateArgsForPullRequestAnalysis
+function UpdateArgsForPullRequestAnalysis($cmdLineArgs, $serviceEndpoint)
 {
-	param([string]$cmdLineArgs)
+    $prcaEnabled = GetTaskContextVariable "PullRequestSonarQubeCodeAnalysisEnabled"
+    if ($prcaEnabled -ieq "true")
+    {
+        if ($cmdLineArgs -and $cmdLineArgs.ToString().Contains("sonar.analysis.mode"))
+        {
+            throw "Error: sonar.analysis.mode seems to be set already. Please check the properties of SonarQube build tasks and try again."
+        }
 
-	$prcaEnabled = GetTaskContextVariable "PullRequestSonarQubeCodeAnalysisEnabled"
-	if ($prcaEnabled -ieq "true")
-	{
-		if ($cmdLineArgs -and $cmdLineArgs.ToString().Contains("sonar.analysis.mode"))
-		{
-			throw "Error: sonar.analysis.mode seems to be set already. Please check the properties of SonarQube build tasks and try again."
-		}
+        Write-Verbose "PullRequestSonarQubeCodeAnalysisEnabled is true, setting command line args for sonar-runner."
+        $sqServerVersion = GetSonarQubeServerVersion $serviceEndpoint.Url
+
+        if (!$sqServerVersion)
+        {
+            #we want to fail the build step if SonarQube server version isn't fetched
+            throw "Error: Unable to fetch SonarQube server version. Please make sure SonarQube server is reachable at $($serviceEndpoint.Url)"
+        }
+
+        Write-Verbose "SonarQube server version:$sqServerVersion"
+
+        #strip out '-SNAPSHOT' if it is present in version (developer versions of SonarQube might return version in this format: 5.2-SNAPSHOT)
+        $sqServerVersion = $sqServerVersion.ToUpper().Replace("-SNAPSHOT", "")
 
-		Write-Verbose "PullRequestSonarQubeCodeAnalysisEnabled is true, setting command line args for incremental mode for sonar-runner..."
+        $sqVersion = New-Object -TypeName System.Version -ArgumentList $sqServerVersion
+        $sqVersion5dot2 = New-Object -TypeName System.Version -ArgumentList "5.2"
 
-		$cmdLineArgs = $cmdLineArgs + " " + "/d:sonar.analysis.mode=incremental"
+        #For SQ version 5.2+ use issues mode, otherwise use incremental mode. Incremental mode is not supported in SQ 5.2+
+        if ($sqServerVersion -ge $sqVersion5dot2)
```

Comment:
I assume that -ge ends up calling the CompareTo method off the Version object? Could you add a comment about this?